### PR TITLE
IRIS-3916 | add test case that covers resetting password for username with spaces

### PR DESCRIPTION
**Pull request by tortila**
[Ticket](https://wikia-inc.atlassian.net/browse/IRIS-3916)

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.wikia.webdriver.testcases.auth;
 
 import com.wikia.webdriver.common.core.EmailUtils;
+import com.wikia.webdriver.common.core.annotations.InBrowser;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.annotations.InBrowser'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
 import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.drivers.Browser;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.drivers.Browser'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java [hunk 3]
```diff
@@ -64,4 +57,20 @@ public void anonCanRemindPasswordOnUserLoginSpecialPageUsingLowerCaseUserName()
 
     assertTrue(resetPass.newPasswordSetSuccessfully());
   }
+
+  private void performResetPasswordFlowForUser(User user) {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java]: `executeResetPasswordFlow`?

File: src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java [hunk 4]
```diff
@@ -20,24 +20,17 @@
...
-
-    String resetLink = base.getPasswordResetLink(credentials.email, credentials.emailPassword);
-    ResetPasswordPage resetPass = new ResetPasswordPage(resetLink);
-    resetPass.setNewPassword(User.FORGOTTEN_PASSWORD.getPassword());
+    performResetPasswordFlowForUser(User.FORGOTTEN_PASSWORD);
+  }
 
-    assertTrue(resetPass.newPasswordSetSuccessfully());
+  @Test
+  public void anonCanResetPasswordForUsernameWithSpace() {
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/testcases/auth/ForgottenPasswordTests.java]: Nitpick: the configured username has more than one space. 🐳

**Pull request merged**